src/cdk/overlay/dispatchers/overlay-outside-click-dispatcher.spec.ts

```diff
@@ -155,6 +155,40 @@ describe('OverlayOutsideClickDispatcher', () => {
 
     overlayRef.dispose();
   });
+
+  it(
+    'should not throw an error when when closing out related components via the' +
+      ' outsidePointerEvents emitter on background click',
+    fakeAsync(() => {
+      const firstOverlayRef = overlay.create();
+      firstOverlayRef.attach(new ComponentPortal(TestComponent));
+      const secondOverlayRef = overlay.create();
+      secondOverlayRef.attach(new ComponentPortal(TestComponent));
+      const thirdOverlayRef = overlay.create();
+      thirdOverlayRef.attach(new ComponentPortal(TestComponent));
+
+      const spy = jasmine.createSpy('background click handler spy').and.callFake(() => {
+        // we close out both overlays from a single outside click event
+        firstOverlayRef.detach();
+        thirdOverlayRef.detach();
+      });
+      firstOverlayRef.outsidePointerEvents().subscribe(spy);
+      secondOverlayRef.outsidePointerEvents().subscribe(spy);
+      thirdOverlayRef.outsidePointerEvents().subscribe(spy);
+
+      const backgroundElement = document.createElement('div');
+      document.body.appendChild(backgroundElement);
+
+      expect(() => backgroundElement.click()).not.toThrowError();
+
+      expect(spy).toHaveBeenCalled();
+
+      backgroundElement.parentNode!.removeChild(backgroundElement);
```

Comment:
Should this go in an afterEach block in a separate describe block, if it needs to be run even if the test fails?

Reply:
The jasmine tests don't exit on a failed expect so the closing logic will run regardless.